hw/ip/flash_ctrl/data/flash_ctrl.sv.tpl

```diff
@@ -530,7 +530,7 @@ module flash_ctrl
   // a read request is seen from software but a read operation is not enabled
   // AND there are no pending entries to read from the fifo.
   logic rd_no_op_d, rd_no_op_q;
-  assign rd_no_op_d = rd_fifo_ren & ~rd_op_valid & ~sw_rvalid;
+  assign rd_no_op_d = rd_fifo_ren & ~rd_op_valid & ~sw_rvalid & sw_sel;
```

Comment:
Random drive-by suggestion: We'll probably want to update the comment above explaining that `sw_sel` qualifies that `rd_op_valid` and `sw_rvalid` are ignored while arbitration has not selected the software side (i.e. during initialization and transactions from the hardware interface, both of which are guaranteed to complete(?)), so `sw_sel` _will_ eventually go high and activate looking at the error conditions.

The statement made by this logic depends a little on the particular behavior of the FSM inside flash_ctrl_arb.

Reply:
updated.